x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -588,6 +590,30 @@ private static void validateInExpression(LogicalPlan p, Set<Failure> localFailur
                 In.class));
     }
 
+    private static void validateConditional(LogicalPlan p, Set<Failure> localFailures) {
+        p.forEachExpressions(e ->
+            e.forEachUp((ConditionalFunction greatest) -> {
+                    int idx = 0;
+                    DataType dt = DataType.NULL;
+                    for (; idx < greatest.children().size(); idx++) {
```

Comment:
Maybe add a one line comment here about this loop and the next one? (iterating until the first non-null value is found then continue and check for types equality)